Beta_Integrations/integration-Vertica.yml

```diff
@@ -0,0 +1,178 @@
+        """
+        if isinstance(v, datetime.datetime):
+            return v.strftime("%Y-%m-%dT%H:%M:%S")
+        return v
+
+
+    def connect_db():
+        try:
+            connection = vertica_python.connect(**DB_PARAMS)
+        except vertica_python.errors.ConnectionError:
+            return_error('Could not connect to DB, please re-check DB params.')
+        return connection
+
+
+    ''' COMMANDS + QUERY FUNCTIONS '''
+
+    def test_module():
+        """
+        Performs basic query on default system tables
+        """
+        cursor.execute("SELECT * FROM system_tables ORDER BY table_schema, table_name LIMIT 2;")
+        rows = cursor.fetchall()
+        if cursor.rowcount == 0:
+            return_error('No results were returned from the DB.')
+        demisto.results('ok')
+
+
+    def query_command():
+        """
+        Execute a query against the DB
+        """
+        # Init main vars
+        contents = []
+        context = {}
+        title = 'Vertica Query'
+
+        # Get arguments from user
+        query = demisto.args().get('query')
+        limit = int(demisto.args().get('limit', 50))
+        # Query and get raw response (list of ordered dicts)
+        rows = query_request(query)
+
+        # Parse response into context & content entries
+        if rows:
+            if limit:
+                rows = rows[:limit]
+            title = 'Vertica Query Results:'
+
+            for i, row in enumerate(rows):
+                rows[i] = { underscoreToCamelCase(k): convert_datetime_to_string(v) for k, v in row.items() }
+
+            contents = rows
+            context['Vertica.Query(val.Query && val.Query === obj.Query)'] = {
+                'Query': query,
+                'Rows': rows
+            }
+
+        demisto.results({
+            'Type': entryTypes['note'],
+            'ContentsFormat': formats['json'],
+            'Contents': contents,
+            'ReadableContentsFormat': formats['markdown'],
+            'HumanReadable': tableToMarkdown(title, contents, removeNull=True),
+            'EntryContext': context
+        })
+
+
+    def query_request(query):
+        try:
+            cursor.execute(query)
+        except vertica_python.errors.MissingRelation:
+            return_error('Error while executing query.')
+        rows = cursor.fetchall()
+        if cursor.rowcount in {0, -1}:
+            return False
+        else:
+            return rows
+
+
+    ''' COMMANDS MANAGER / SWITCH PANEL '''
+
+    LOG('Command being called is %s' % (demisto.command()))
+
+    try:
+        connection = connect_db()
+        cursor = connection.cursor('dict')
+        if demisto.command() == 'test-module':
+            test_module()
+        elif demisto.command() == 'vertica-query':
+            query_command()
+    # Log exceptions
+    except Exception as e:
+        LOG(e)
+        LOG.print_log()
+        raise
+    finally:
+        if 'connection' in globals():
```

Comment:
in which case this condition will be `False` ?

Reply:
In case connection var failed to instantiate properly, aka no connection was formed and Exception was thrown as result.